### DISCUSSION TITLE
wireguard

### DISCUSSION THREAD

**Discussion by titof2375** [tteck/Proxmox#138] [General]
hello, I just put back in place the wirguard, which does not work when I am in 4 G. I see that in your script you do not need to put a domain name for the vpn to work. So I wonder how it will work when the public IP is changed?

**Comment by ianalexis** [comment]
### TL:DR
The script is not that useful...
Install **PiVPN** in a **LXC** without GUI and open port **51820** to it.
**---------------------------------------------------------------**
### The script needs some changes to really work.
1) In the install ask for a MacAddres wit a fix ip and port 51820 open.
Ask for the following values to edit the PiVPN config /etc/pivpn/wireguard/setupVars.conf file with:
pivpnDNS1=[PiHole IP if you have or a DNS ip]
pivpnDNS2=[Another DNS ip]
pivpnHOST=[Domain]

2) Use the values to WGDashborad (becouse i dont know why it uses a different config) wg-dashboard.ini (forgot where is it):
peer_global_dns=(same as pivpnDNS1)
remote_endpoint=(same as pivpnHOST)

**WGDashborad dosnt have a DNS backup so if your DNS fails the connection will fail.**
**---------------------------------------------------------------**
### In my case:
The solution was install a clean debian LXC and then WireGuard with pivpn (same as used here).
A good tutorial [https://www.wundertech.net/setup-wireguard-on-a-raspberry-pi-vpn-setup-tutorial/](url)
The problem is i cant find a good GUI but i use:
pivpn -a,  add              Create a client conf profile"
pivpn -c,  clients          List any connected clients to the server"
pivpn -d,  debug            Start a debugging session if having trouble"
pivpn -l,  list             List all clients"
pivpn -qr, qrcode           Show the qrcode of a client for use with the mobile app"
pivpn -r,  remove           Remove a client"
pivpn -h,  help             Show this help dialog"
pivpn -u,  uninstall        Uninstall pivpn from your system!"
pivpn -up, update           Updates PiVPN Scripts"
pivpn -bk, backup           Backup VPN configs and user profiles"

**Reply by ianalexis** [reply]
@titof2375 Will help a thumb up here https://github.com/tteck/Proxmox/discussions/383#discussion-4263021
